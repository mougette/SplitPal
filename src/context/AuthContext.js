import React, { useState} from "react";
import createDataContext from './createDataContext';
import {Post} from '../components/RestPost';
import {Get} from '../components/RestGet';

const authReducer = (state, action) => {
  switch (action.type) {
    case 'signout':
      return {token: null, email: ''};
    case 'signin':
    return {
            email: action.payload.email,
            token: action.payload.token,
          };
    case 'signup':
      return {
        token: action.payload.token,
        email: action.payload.email,
      };
  }
};

const signup = dispatch => {
    const [account, setAccount] = useState("");
  return ({email, password, password2, firstName, lastName, phoneNumber}) => {
    Post("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login",
                JSON.stringify({
                 email: email,
                 password: password,
                 password2: password2,
                    firstName: firstName,
                    lastName: lastName,
                    phoneNumber: phoneNumber,
                 }))
        .then(response => {
    console.log("Account: " + account + " Account End");
      if(response == "Account Successfully Made") {
              dispatch({
                  type: 'signup',
                  payload: {
                      token: 'some access token here',
                      email: email,
                  },
              });
      }
      else {
              alert(response)
      }
      }
      );


  };
};

const signin = dispatch => {
  return ({email, password}) => {
    // Do some API Request here
   Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login","?email="+email+"&password="+password)
   .then(response =>{
    if(response == "User info correct"){
    dispatch({
      type: 'signin',
      payload: {
        token: 'some access token here',
        email: email,
      },
    });
    }});
  };
};

const signout = dispatch => {
  return () => {
    dispatch({type: 'signout'});
  };
};

export const {Provider, Context} = createDataContext(
  authReducer,
  {signin, signout, signup},
  {token: null, email: ''},
);
