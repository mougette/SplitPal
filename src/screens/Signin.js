import React, {useState, useContext} from 'react';
import {Icon} from 'react-native-elements';
import {View, Text, StyleSheet, TouchableOpacity, Keyboard, TouchableWithoutFeedback} from 'react-native';
import {Input, Button} from 'react-native-elements';
import {Context as AuthContext} from '../context/AuthContext';
import SplitPalLogoComponent from '../components/SplitPalLogoComponent';

const Signin = ({navigation}) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const {state, signin} = useContext(AuthContext);

  return (
    <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
      <View style={styles.master}>

        <SplitPalLogoComponent />
        <Input
          placeholder="Email"
          onChangeText={setEmail}
          value={email}
          leftIcon={<Icon name="envelope" type="font-awesome" size={24} />}
          textContentType="emailAddress"
          keyboardType="email-address"
          autoCapitalize="none"
          autoCorrect={false}
        />
        <Input
          placeholder="Password"
          onChangeText={setPassword}
          value={password}
          leftIcon={<Icon name="key" type="font-awesome" size={24} />}
          secureTextEntry
          autoCapitalize="none"
          autoCorrect={false}
        />
        <Button
          title="Login"
          type="clear"
          onPress={() => {
            signin({email, password});
          }}
        />
        <View style={styles.link}>
          <Text style={styles.text}>Dont have an account? </Text>
          <TouchableOpacity onPress={() => navigation.navigate("Signup") }>
            <Text style={styles.blueText}>Sign up Here.</Text>
          </TouchableOpacity>
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
};

const styles = StyleSheet.create({
  master: {
    padding: 16,
    flex: 1,
    alignItems: 'stretch',
    justifyContent: 'center',
  },
  header: {
    fontSize: 32,
    marginBottom: 18,
    alignSelf: 'center',
  },
  text: {
    fontSize: 16,
    marginTop: 16,
  },
  blueText: {
    fontSize: 16,
    marginTop: 16,
    color: '#349beb',
  },
  link: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
});

export default Signin;
