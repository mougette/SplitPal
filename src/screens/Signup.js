import React, {useState, useContext} from 'react';
import {Icon} from 'react-native-elements';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';
import {Input, Button} from 'react-native-elements';
import {Context as AuthContext} from '../context/AuthContext';
import SplitPalLogoComponent from '../components/SplitPalLogoComponent';

const Signup = ({navigation}) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [password2, setPassword2] = useState('');
  const {state, signin} = useContext(AuthContext);

  return (
    <View style={styles.master}>

      <SplitPalLogoComponent />
      <Input
        placeholder="Email"
        onChangeText={setEmail}
        value={email}
        leftIcon={<Icon name="envelope" type="font-awesome" size={24} />}
        textContentType="emailAddress"
        autoCapitalize="none"
        autoCorrect={false}
      />
      <Input
        placeholder="Password"
        onChangeText={setPassword}
        value={password}
        leftIcon={<Icon name="key" type="font-awesome" size={24} />}
        secureTextEntry
        autoCapitalize="none"
        autoCorrect={false}
      />
      <Input
        placeholder="Confirm Password"
        onChangeText={setPassword2}
        value={password2}
        leftIcon={<Icon name="key" type="font-awesome" size={24} />}
        secureTextEntry
        autoCapitalize="none"
        autoCorrect={false}
      />
      <Button
        title="Login"
        type="clear"
        onPress={() => {
          signin({email, password});
        }}
      />
      <View style={styles.link}>
        <Text style={styles.text}>Dont have an account? </Text>
        <TouchableOpacity onPress={() => navigation.navigate("Signin")}>
          <Text style={styles.blueText}>Sign up Here.</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  master: {
    padding: 16,
    flex: 1,
    alignItems: 'stretch',
    justifyContent: 'center',
  },
  header: {
    fontSize: 32,
    marginBottom: 18,
    alignSelf: 'center',
  },
  text: {
    fontSize: 16,
    marginTop: 16,
  },
  blueText: {
    fontSize: 16,
    marginTop: 16,
    color: '#349beb',
  },
  link: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
});

export default Signup;
