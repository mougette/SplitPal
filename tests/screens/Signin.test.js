import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react-native'
import {Provider as AuthProvider} from '../../src/context/AuthContext.js';
import {Context as AuthContext} from '../../src/context/AuthContext';
import {navigate} from '@reach/router'
import Signin from '../../src/screens/Signin'
import Signup from '../../src/screens/Signup'
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ state: { email: "e@something.com", token: "something here" } }),
  })
  );
  const createTestProps = (props: Object) => ({
    navigation: {
      navigate: jest.fn()
    },
    ...props
  });
describe('Test Hooks', () => {
    const test = jest.fn();
    const Stack = createBottomTabNavigator();
    it('Login', () => {
    const {getByText, asFragment} = render(
    <AuthProvider>
    <NavigationContainer>
    <Stack.Navigator>
    <Stack.Screen name="Signin" component={Signin} />
    <Stack.Screen name="Signup" component={Signup} />
    </Stack.Navigator>
    </NavigationContainer>
    </AuthProvider>);
    fireEvent.press(getByText("Login"));
    expect(asFragment).toMatchSnapshot()
    });
    it('SignUp', () => {
        const {getByText, asFragment} = render(
        <AuthProvider>
        <NavigationContainer>
        <Stack.Navigator>
        <Stack.Screen name="Signin" component={Signin} />
        <Stack.Screen name="Signup" component={Signup} />
        </Stack.Navigator>
        </NavigationContainer>
        </AuthProvider>);
        fireEvent.press(getByText("Sign up Here."));
        expect(asFragment).toMatchSnapshot()
        })


});