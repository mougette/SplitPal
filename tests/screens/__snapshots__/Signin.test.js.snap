// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Hooks Login 1`] = `undefined`;

exports[`Test Hooks SignUp 1`] = `undefined`;
